armi/reactor/reactors.py

```diff
@@ -293,8 +293,9 @@ def powerMultiplier(self):
     def lib(self):
         """"Get the microscopic cross section library."""
         if self._lib is None:
-            runLog.info("Loading microscopic cross section library ISOTXS")
-            self._lib = nuclearDataIO.ISOTXS()
+            runLog.warning(
+                f"A cross section library has not yet been assigned to {self}."
+            )
```

Comment:
I agree that this is an important capability, but attaching the behavior loading an ISOTXS to a getLib function is a bit much.  Just pointing to that nuclearDataIO.ISOTXS() is a function that reads the file name "ISOTXS" unless a different kwarg is passed.

There are also other types of lib formats possible besides isotxs. I really like that Jake is making the getter a simple getter rather than a getter + read if it is None.

Maybe the warning message should suggest a way for the user to set up lib?

Reply:
@onufer, I think @ntouran, @youngmit and I have decided to maintain the functionality but I am trying to make it more explicit from a user perspective versus an automatic ISOTXS load when it's "needed". 

@ntouran also mentioned to me that ISOTXS is the "defacto standard" of ARMI, so I think that this should also be documented.

Reply:
Yeah, I get that loading and ISOTXS should be easy, but I have heard Nick say methods should do one thing. I am not suggesting removing the functionality, but just separating it from the getter. Anyway, it sounds like you have already decided.